include/RAJA/pattern/tensor/internal/ET/MultiplyOperator.hpp

```diff
@@ -375,7 +381,305 @@ namespace expt
 
       }
 
-    };
+
+      template<typename T>
+      struct Diag{
+          static_assert(!std::is_same<T,void>::value,"diag");
+      };
+
+      template<typename I, TensorTileSize TTS, typename B, typename S>
+      struct Diag< StaticTensorTile<I,TTS,B,S> >{
+          static_assert(std::is_same<I,void>::value,"diag");
+      };
+
+      template<typename STORAGE, typename TILE_TYPE, typename INDEX>
+      struct MultiplyBridge {
+
+          Diag<TILE_TYPE> diag;
+
+          RAJA_INLINE
+          RAJA_HOST_DEVICE
+          static
+          void multiply_into_result(STORAGE &result, TILE_TYPE const &tile, LEFT_OPERAND_TYPE const &et_left, RIGHT_OPERAND_TYPE const &et_right)
+          {
+            //using LHS_STORAGE = typename LEFT_OPERAND_TYPE::result_type;
+    
+            // get tile size from matrix type
+            auto tile_size = left_type::result_type::s_dim_elem(1);
+            auto k_size = et_left.getDimSize(1);
+            // TODO: check that left and right are compatible
+            // m_left.getDimSize(1) == m_right.getDimSize(0)
```

Comment:
Don't we have a flag to enable indix checking, we could have optional asserts that used the same of a similar compile flag.